src/OpenLoco/src/Ui/Widgets/TabWidget.cpp

```diff
@@ -0,0 +1,66 @@
+#include "TabWidget.h"
+#include "Graphics/SoftwareDrawingEngine.h"
+#include "Graphics/TextRenderer.h"
+#include "Localisation/Formatting.h"
+#include "Ui/Window.h"
+
+namespace OpenLoco::Ui::Widgets
+{
+    // 0x004CADE8
+    static void drawTabBackground(Gfx::DrawingContext& drawingCtx, const Widget& widget, const WidgetState& widgetState)
+    {
+        auto* window = widgetState.window;
+        Ui::Point placeForImage(widget.left + window->x, widget.top + window->y);
+
+        // TODO: This is always ImageIds::tab at the moment, we should make this implicit.
+        ImageId imageId = ImageId{ widget.image };
+        if (widgetState.activated)
+        {
+            // TODO: remove image addition
+            imageId = imageId.withIndexOffset(1);
+        }
+
+        auto colour = widgetState.colour;
+        if (widgetState.disabled)
+        {
+            uint8_t c;
+            if (colour.isTranslucent())
+            {
+                c = Colours::getShade(colour.c(), 4);
+                drawingCtx.drawImageSolid(placeForImage + Ui::Point{ 1, 1 }, imageId, c);
+                c = Colours::getShade(colour.c(), 2);
+                drawingCtx.drawImageSolid(placeForImage, imageId, c);
+            }
+            else
+            {
+                c = Colours::getShade(colour.c(), 6);
+                drawingCtx.drawImageSolid(placeForImage + Ui::Point{ 1, 1 }, imageId, c);
+                c = Colours::getShade(colour.c(), 4);
+                drawingCtx.drawImageSolid(placeForImage, imageId, c);
+            }
+
+            return;
+        }
+
+        imageId = imageId.withPrimary(colour.c());
+
+        drawingCtx.drawImage(placeForImage, imageId);
+    }
+
+    void Tab::draw(Gfx::DrawingContext& drawingCtx, const Widget& widget, const WidgetState& widgetState)
+    {
+        if (widget.content == Widget::kContentNull)
+        {
+            return;
+        }
+
+        if (widgetState.disabled)
+        {
+            return;
+        }
```

Comment:
I don't think it was ever possible to draw disabled tabs, the drawing code however checks for this state but that code was also shared with other widgets. I'll have to see what it looks like if I remove this.

Reply:
So removing that line and disabling one tab in the demo window looks like following:
![image](https://github.com/user-attachments/assets/ccb0c1eb-d664-4bfe-8e0d-3129fb51f53a)
I think this got to be an oversight given that in Widget.cpp the function drawTabImpl will actually just return when its disabled, it works drawing wise.

Reply:
Yeah, basically, tabs don't get into this state in the original game — they're either enabled or hidden completely.

In OpenRCT2, we did have a need for a disabled tab sprite at some point. As 'disabled' images paint all non-transparent pixels as a darker colour, we add an alternate sprite for the one case we needed it for. However, because it was unintuitive, we removed it again later.

I think your code works fine, basically, and we can leave it as-is.